python/ray/autoscaler/batching_node_provider.py

```diff
@@ -230,6 +249,17 @@ def terminate_node(self, node_id: str) -> Optional[Dict[str, Any]]:
                 f"{node_type}. Skipping termination request."
             )
 
+        # Terminate node
         self.scale_request.desired_num_workers[node_type] -= 1
         self.scale_request.workers_to_delete.add(node_id)
+
+        # Scale down all nodes in replica if node_id is part of a multi-host podslice
+        node_replica_index = self.node_data_dict[node_id].replica_index
+        if node_replica_index is not None:
+            for worker_id in self.replicas_to_nodes[node_replica_index]:
+                # Check if worker has already been scheduled to delete
+                if worker_id not in self.scale_request.workers_to_delete:
+                    # Assume all workers in a group are of the same type
+                    self.scale_request.desired_num_workers[node_type] -= 1
```

Comment:
Do we need to update `self.scale_request.desired_num_workers[node_type]` for every Pod in a PodSlice?

Reply:
https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/autoscaler/_private/kuberay/node_provider.py?L461

Reply:
Done in [a06581c](https://github.com/ray-project/ray/pull/43470/commits/a06581c75fbfae32ec5b237c4b40f2aee28af7a1).